inc/RestApi/PluginsController.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Bluehost\Maestro\RestApi;
+
+use Exception;
+use WP_REST_Server;
+use WP_REST_Response;
+
+use Bluehost\Maestro\Plugin;
+use Bluehost\Maestro\WebPro;
+
+/**
+ * Class PluginsController
+ */
+class PluginsController extends \WP_REST_Controller {
+
+	/**
+	 * The namespace of this controller's route.
+	 *
+	 * @since 1.1.1
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'bluehost/maestro/v1';
+
+	/**
+	 * The current Web Pro accessing the endpoint
+	 *
+	 * @since 1.1.1
+	 *
+	 * @var WebPro
+	 */
+	private $webpro;
+
+	/**
+	 * Registers the Plugins routes
+	 *
+	 * @since 1.1.1
+	 */
+	public function register_routes() {
+
+		register_rest_route(
+			$this->namespace,
+			'/plugins',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_plugins' ),
+					'permission_callback' => array( $this, 'check_permission' ),
+				),
+			)
+		);
+
+	}
+
+	/**
+	 * Callback for the plugins get endpoint
+	 *
+	 * Returns a list of installed plugins with details and updates
+	 *
+	 * @since 1.1.1
+	 *
+	 * @return WP_Rest_Response Returns a standard rest response with a list of plugins
+	 */
+	public function get_plugins() {
+		if ( ! function_exists( 'get_plugin_data' ) ) {
+			include_once ABSPATH . 'wp-admin/includes/plugin.php';
+		}
+
+		if ( ! function_exists( 'get_option' ) ) {
+			include_once ABSPATH . 'wp-includes/options.php';
+		}
+
+		// Make sure we populate the plugins updates transient
+		wp_update_plugins();
+
+		$installed_plugins = get_plugins();
+		$updates           = get_site_transient( 'update_plugins' );
+		$auto_updates      = get_option( 'auto_update_plugins' );
+		$plugins           = array();
+
+		foreach ( $installed_plugins as $plugin_slug => $plugin_details ) {
+			$plugin = new Plugin( $plugin_slug, $updates, $plugin_details, $auto_updates );
```

Comment:
I think it would make more sense to pass only the data specific to each plugin here for `$updates` and `$auto_updates` rather than passing the entire array of data and having logic within the `Plugin` class to figure that out. I think that the names could be more descriptive as well. Maybe `$available_update` and `$auto_updates_enabled`.

Reply:
That makes sense, I will refactor to take this into account and filter and pass only the specific plugin's details.

Reply:
Yes, might have slipped between the to and fro of controllers to register_rest_field and back.
Done now, refactored this to work with details of a specific plugin